cmake/reports/CMakeLists.txt

```diff
@@ -16,3 +16,13 @@ foreach(report ram_report rom_report)
     DEPENDS ${logical_target_for_zephyr_elf}
     )
 endforeach()
+
+add_custom_command(
+  TARGET ${IMAGE}rom_report
+  POST_BUILD
+  COMMAND
+  ${PYTHON_EXECUTABLE}
+  ${ZEPHYR_BASE}/scripts/partition_manager_report.py
+  --input ${PROJECT_BINARY_DIR}/include/generated/partitions.yml
+  "$<$<NOT:$<TARGET_EXISTS:partition_manager>>:--quiet>"
```

Comment:
I believe you mentioned one could have empty COMMAND's.

If so, let's put the command in a variable and do

"$<$<TARGET_EXISTS:partition_manager>:${the_command}>"

Then we can avoid invoking partition_manager_report.py for non-PM builds.

Reply:
It seems this is not really possible to do in a good way.